crates/rattler_networking/src/gcs_middleware.rs

```diff
@@ -63,3 +63,37 @@ async fn authenticate_with_google_cloud(mut req: Request) -> MiddlewareResult<Re
         Err(e) => Err(reqwest_middleware::Error::Middleware(anyhow::Error::new(e))),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use reqwest::Client;
+    use tempfile;
+
+    #[tokio::test]
+    async fn test_gcs_middleware() {
```

Comment:
Should we only run this test if this variable is available?

Reply:
We only run this test when the `gcs` feature is enabled. So that already makes the test run pretty seldomly. I havent' found a way to print a big fat warning, so I prefer to fail the test.